pkg/volume/csi/testing/testing.go

```diff
@@ -52,7 +52,17 @@ func NewTestPlugin(t *testing.T, client *fakeclient.Clientset) (*volume.VolumePl
 	factory := informers.NewSharedInformerFactory(client, csi.CsiResyncPeriod)
 	csiDriverInformer := factory.Storage().V1().CSIDrivers()
 	csiDriverLister := csiDriverInformer.Lister()
-	go factory.Start(wait.NeverStop)
+
+	factory.Start(wait.NeverStop)
+	syncedTypes := factory.WaitForCacheSync(wait.NeverStop)
+	if len(syncedTypes) != 1 {
+		t.Fatalf("informers are not synced")
+	}
+	for ty, ok := range syncedTypes {
+		if !ok {
+			t.Fatalf("failed to sync: %#v", ty)
+		}
+	}
 
 	host := volumetest.NewFakeVolumeHostWithCSINodeName(t,
```

Comment:
Only the `AttachDetachVolumeHost` has a `volumeAttachmentLister`. This test plugin constructor is not used in the CSI attacher test and doesn't use that volume host interface.

Reply:
Oh I see. Looks like there is quite a bit of code change in this area recently. I was referring to slightly older code.

Isn't AttachDetachVolumeHost using the fakeVolumeHost here? What am I missing?
https://github.com/kubernetes/kubernetes/blob/0eea524c28ea4c0e4a0d2746184f3dc63ff124b9/pkg/volume/testing/volume_host.go#L242

Reply:
It is, and the `fakeVolumeHost` has a field for the `volumeAttachmentLister`.

The reason for this splitting-out of different volume hosts was because previously the `fakeVolumeHost` type implemented both the `AttachDetachVolumeHost` and `KubeletVolumeHost` interfaces, which made it not possible to test behavior differences between the two, because it passes both conditionals here.
https://github.com/kubernetes/kubernetes/blob/9d8f994d4ec15a5f37381a5d8ec37c6c53f81543/pkg/volume/csi/csi_plugin.go#L193-L216